src/commands/sftp-user/create.tsx

```diff
@@ -0,0 +1,119 @@
+import { ExecRenderBaseCommand } from "../../lib/basecommands/ExecRenderBaseCommand.js";
+import {
+  makeProcessRenderer,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { ReactNode } from "react";
+import { assertStatus } from "@mittwald/api-client-commons";
+import { Success } from "../../rendering/react/components/Success.js";
+import { Value } from "../../rendering/react/components/Value.js";
+import { projectFlags } from "../../lib/resources/project/flags.js";
+import { MittwaldAPIV2Client } from "@mittwald/api-client";
+import { expireFlags } from "../../lib/flags/expireFlags.js";
+import { sftpUserFlagDefinitions } from "../../lib/resources/sftp/flags.js";
+
+type Result = {
+  sftpUserId: string;
+};
+
+type SftpUserCreationPayload = Parameters<
+  MittwaldAPIV2Client["sshsftpUser"]["sftpUserCreateSftpUser"]
+>[0]["data"];
+
+export class Create extends ExecRenderBaseCommand<typeof Create, Result> {
+  static summary = "Create a new SFTP user";
+  static flags = {
+    ...projectFlags,
+    ...processFlags,
+    ...expireFlags("SFTP User", false),
+    description: sftpUserFlagDefinitions.description({ required: true }),
+    "public-key": sftpUserFlagDefinitions["public-key"]({
+      exactlyOne: ["public-key", "password"],
+    }),
+    password: sftpUserFlagDefinitions.password({
+      exactlyOne: ["public-key", "password"],
+    }),
+    "access-level": sftpUserFlagDefinitions["access-level"](),
+    directories: sftpUserFlagDefinitions.directories({ required: true }),
+  };
+
+  protected async exec(): Promise<Result> {
+    const process = makeProcessRenderer(this.flags, "Creating a new SFTP User");
+    const projectId = await this.withProjectId(Create);
+    const {
+      description,
+      "public-key": publicKey,
+      password,
+      expires,
+      "access-level": accessLevel,
+      directories,
+    } = this.flags;
+
+    let authentication: SftpUserCreationPayload["authentication"];
+    if (password) {
+      authentication = { password };
+    } else if (publicKey) {
+      authentication = {
+        publicKeys: [{ comment: "Public key set through CLI", key: publicKey }],
+      };
+    } else {
+      throw new Error("The authentication method could not be set correctly.");
+    }
+
+    const sftpUserCreationPayload: SftpUserCreationPayload = {
+      authentication,
+      description,
+      directories: [directories[0], ...directories.slice(1)],
```

Comment:
Is this actually required, because the `SftpUserCreationPayload` is typed this way? Or would just `directories` also work?

Reply:
If there is a different way, I haven't found it.
directories is typed very specifically as `directories: [string, ...string[]];` wich requires me to pass in a string first,and then unravel the rest of the string[]

Reply:
Yeah, I figured as much. Oh, well... 🤷